File: Omkar's Pet Cat/gemini1.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

void solve() {
    int n;
    cin >> n;
    vector<int> a(n);
    int min_val = 2e9 + 7;
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
        min_val = min(min_val, a[i]);
    }
    
    int count = 0;
    for (int i = 0; i < n; ++i) {
        if (a[i] == min_val) {
            count++;
        }
    }
    
    if (count == 1) {
        cout << "YES\n";
        return;
    }
    
    for (int i = 0; i < n; ++i) {
        if (a[i] % min_val != 0) {
            cout << "YES\n";
            return;
        }
    }
    cout << "NO\n";
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t;
    if (cin >> t) {
        while (t--) {
            solve();
        }
    }
    return 0;
}

File: Omkar's Pet Cat/gemini1.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    t = int(input_data[0])
    idx = 1
    results = []
    
    for _ in range(t):
        n = int(input_data[idx])
        idx += 1
        
        a = [int(x) for x in input_data[idx : idx + n]]
        idx += n
        
        min_val = min(a)
        count = a.count(min_val)
        
        if count == 1:
            results.append("YES")
        else:
            possible = False
            for x in a:
                if x % min_val != 0:
                    possible = True
                    break
            
            if possible:
                results.append("YES")
            else:
                results.append("NO")
                
    sys.stdout.write("\n".join(results) + "\n")

if __name__ == '__main__':
    solve()